client/src/pages/admin/Index.js

```diff
@@ -30,8 +34,26 @@ const GQL_SAVE_ADMIN_SETTINGS = gql`
     saveAdminSettings(settings: $settings)
   }
 `
-
+const CLEAR_CACHE = gql`
+  mutation {
+    clearCache
+  }
+`
+const RELOAD_DICTIONARY = gql`
+  mutation {
+    reloadDictionary
+  }
+`
+const USER_ACTIVITIES = gql`
+  query {
+    userActivities
+  }
+`
 const BaseAdminIndex = ({ pageDispatchers, loadAppData }) => {
+  const [userActivities, setUserActivities] = useState(null)
+  const [recentUsers, setRecentUsers] = useState(null)
+  const [actionLoading, setActionLoading] = useState(false)
```

Comment:
Can't we use the loading mechanism of the boilerplate?

Reply:
Since `recentUsers` and `userActivities` are loaded by demand (as response to user action/button click) by using `API.query` (which returns a Promise) I don't think we can't use `useBoilerplate` hook here.
The reason I prefer to load them by demand is to avoid unnecessary data fetch in case the user opens the page for another purpose than seeing this data. Also, even if we load the data when the page is opened, we will still need a re-fetch/refresh functionality which becomes already available in this way.